x-pack/legacy/plugins/maps/public/angular/map_controller.js

```diff
@@ -124,6 +126,75 @@ app.controller('GisMapController', ($scope, $route, kbnUrl, localStorage, AppSta
     globalState: globalState,
   });
 
+  /* Saved Queries */
+  $scope.showSaveQuery = capabilities.get().maps.saveQuery;
+
+  $scope.$watch(() => capabilities.get().maps.saveQuery, (newCapability) => {
+    $scope.showSaveQuery = newCapability;
+  });
+
+  $scope.onQuerySaved = savedQuery => {
+    $scope.savedQuery = savedQuery;
+  };
+
+  $scope.onSavedQueryUpdated = savedQuery => {
+    $scope.savedQuery = { ...savedQuery };
+  };
+
+  $scope.onClearSavedQuery = () => {
+    delete $scope.savedQuery;
+    delete $state.savedQuery;
+    onQueryChange({
+      filters: [],
+      query: {
+        query: '',
+        language: localStorage.get('kibana.userQueryLanguage')
+      },
+    });
+  };
+
+  function updateStateFromSavedQuery(savedQuery) {
+    if (savedQuery.attributes.timefilter) {
+      if (savedQuery.attributes.timefilter.refreshInterval) {
+        $scope.refreshConfig.interval = savedQuery.attributes.timefilter.refreshInterval.value;
```

Comment:
Need to call `$scope.onRefreshChange`  instead of setting `$scope.refreshConfig` directly.

There are no watches on `$scope.refreshConfig` so setting `$scope.refreshConfig` does not trigger the dispatch of `setRefreshConfig` causing the redux store to never get the updated refresh configuration

Reply:
I have made the change [here](https://github.com/elastic/kibana/pull/44442/commits/4333b872134a12beb57f8e6648e9b624e3398b48#diff-8f39ea21edecd35fad6f4c0df45681a3R159).